### PR TITLE
Switch to using a real csv parser

### PR DESCRIPTION
**Pull request by Lenbok**
There are a couple of emacs CSV parsing libraries that seem to handle quoted fields and embedded commas. This PR plugs in one of those.

(BTW - the "ignoring empty prompts" filter of the previous impl doesn't seem needed, so left it out as YAGNI)

Fixes: #126

### CODE REVIEW

File: chatgpt-shell.el [hunk 1]
```diff
@@ -39,6 +39,7 @@
 (require 'eshell)
 (require 'find-func)
 (require 'ielm)
+(require 'pcsv)
```
Comment (xenodium) [inline chatgpt-shell.el:42]: `pcsv` doesn't look to be a built-in package. We should move `(require 'pcsv)` inside `chatgpt-shell-load-awesome-prompts`, so existing users without this package can still load `chatgpt-shell`.

**Review by xenodium — APPROVED** [review]
Thank you! Left a minor comment for change.

File: chatgpt-shell.el (hunk 1 above)
Comment (Lenbok) [reply chatgpt-shell.el]: Thanks, that's the bit I wasn't sure about where that dep should belong. I guess the requirement should be documented somewhere too?

Comment (xenodium) [reply chatgpt-shell.el]: Thanks for the change. I'll follow up with a small addition (unless you're keen): anyone attempting to use the function without the dependency will be told to install it.

**Pull request merged**